### PR TITLE
#184821806 User should view items

### PR DESCRIPTION
**Pull request by nshimiyejayd**
### What does this PR do?
- This PR implements the way A user can view and interact with items

### Description of Tasks to be completed?
- A user should be able to view all items
- A user  should be able to view single item
- A user can do different operations on the product like wish, add to cart, edit and delete

### How should this be manually tested?
- Clone [repo](https://github.com/atlp-rwanda/vikings-ec-fe.git)
- Change into the directory vikings-ec-fe
- Checkout to the branch: feature/view-products-184821806
- Run the command npm install
- Run the command npm run dev
- After login you can see products
- You should navigate to `/dashboard/products` to view products as seller

### What are the relevant pivotal tracker stories?
[#184821806](https://www.pivotaltracker.com/story/show/184821806) 

### Screenshots
Implemented products view 
image

Implemented single product view 
image

View single product as seller
image

### CODE REVIEW

**Review by ndatumuremyi — APPROVED** [review]

**Review by fabmukunzi — APPROVED** [review]

**Review by irakozetony — APPROVED** [review]

File: src/routes/AppRoutes.js [hunk 1]
```diff
@@ -43,6 +45,16 @@ export const getRoutes = () => [
     <Route path="password_update" element={<UpdatePasswordPage />} />
   </Route>,
   <Route key="key_general_path" path="*" element={<h2>Page Not Found</h2>} />,
+  <>
+    <Route path="/seller-products" element={<Dashboard />} />
+    <Route path="/single-product" element={<SingProductPage />} />
+    <Route key="key__single_product" path="/single-product" element={<HomeLayout />}>
+      <Route index element={<SingProductPage />} />
+    </Route>
```
Comment (ncutixavier) [inline src/routes/AppRoutes.js]: Change the path of the route, 

Get all products: `/products` or `/`
Get single product: `/products/:id`

**Review by ncutixavier — CHANGES_REQUESTED** [review]

File: src/components/products/SellerViewSingleProduct.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    setRating([]);
+    setIReviews(false);
+    axios.get(`${endpoints.product}/${props.product.productId}/ratings`)
+      .then((response) => {
+        response.data.ratings.map((rate) => {
+          setRating((prevRating) => [...prevRating, { ratings: rate.rate }]);
+          rate.buyer.feedback = rate.feedback;
+        });
+        setIReviews(true);
+      })
```
Comment (ncutixavier) [inline src/components/products/SellerViewSingleProduct.js]: You must use redux

File: src/components/products/ProductRatings.js
Comment (ncutixavier) [inline src/components/products/ProductRatings.js:1]: Why didn't you use the package?

File: src/components/products/BuyerViewSingleProduct.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        response.data.ratings.map((rate) => {
+          setRating((prevRating) => [...prevRating, { ratings: rate.rate }]);
+          rate.buyer.feedback = rate.feedback;
+          setReviews((prevReviews) => [...prevReviews, rate.buyer]);
+        });
+        setIReviews(true);
+      })
+      .catch((error) => {
+      });
+  }, [props.products && props.products.id]);
```
Comment (ncutixavier) [inline src/components/products/BuyerViewSingleProduct.js]: You must use redux

File: src/assets/index.css [hunk 4]
```diff
@@ -1,3 +1,9 @@
...
 @tailwind components;
-@tailwind utilities;
\ No newline at end of file
+@tailwind utilities;
+
+.custom-rating {
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+  }
```
Comment (ncutixavier) [inline src/assets/index.css]: You should use tailwind css

**Review by ncutixavier — CHANGES_REQUESTED** [review]

File: src/features/api/customAxios.js [hunk 5]
```diff
@@ -15,7 +15,7 @@ const responseHandler = (response) => response;
 
 const errorHandler = (error) => {
   if (error.response.status === 401) {
-    return (location.href = `${process.env.REACT_APP_BASE_URL}/auth`);
+    return (location.href = `${process.env.FRONT_END_URL}/auth/signin`);
   }
   return Promise.reject(error);
```
Comment (ncutixavier) [inline src/features/api/customAxios.js]: You forgot to clear the local storage. When the token is expired, you need to remove it from local storage and redirect to login

File: src/components/products/ProductRatings.js
Comment (nshimiyejayd) [reply src/components/products/ProductRatings.js:1]: It was harder for me to stylize it and make it working properly

**Review by ncutixavier — APPROVED** [review]

**Pull request merged**